src/com/ppm/integration/agilesdk/connector/jira/JIRAService.java

```diff
@@ -0,0 +1,326 @@
+				JSONObject issue = issues.getJSONObject(i);
+				JSONObject fields = issue.getJSONObject("fields");
+
+				JSONArray worklogs = fields.getJSONObject("worklog").getJSONArray("worklogs");
+				for (int j = 0; j < worklogs.length(); j++) {
+					JSONObject obj = worklogs.getJSONObject(j);
+					Long timeSpentSeconds = obj.getLong("timeSpentSeconds");
+					String started = obj.getString("started");
+					String comment = obj.getString("comment");
+					String self = obj.getString("self");
+					String id = obj.getString("id");
+					String isuSelf = issue.getString("self");
+					String isuId = issue.getString("id");
+
+					// String isuProId = issue.getString("projectId");
+					String isuKey = issue.getString("key");
+					String isuSummary = fields.getString("summary");
+					JIRATempoIssue jti = new JIRATempoIssue(isuSelf, isuId, "", isuKey, 0L, "", isuSummary);
+
+					JIRATempoWorklog jtw = new JIRATempoWorklog(timeSpentSeconds, started, comment, self, id, jti);
+					jtls.add(jtw);
+
+				}
+
+			}
+
+		} catch (JSONException e) {
+
+		}
+
+		return resolveJIRATempoWorklogs(jtls);
+	}
+
+	private JIRAIssue resolveIssue(JSONObject fields, boolean isSubtask, String scheduledStart, String scheduledFinish,
+			String actualFinish) throws JSONException {
+		String name = fields.getString("summary");
+		String status = fields.getJSONObject("status").getString("name");
+		String type = fields.getJSONObject("issuetype").getString("name");
+
+		String total = null;
+		String percentComplete = null;
+		// String timespent = fields.getString("timespent");
+		List<JIRAIssue> children = null;
+		if (!isSubtask) {
+
+			Map<String, String> customfield_10118 = null;
+			if (!"null".equals(fields.get("customfield_10118").toString())) {
+				customfield_10118 = resolveCustomfield_10118(fields.getJSONArray("customfield_10118").getString(0));
+				scheduledStart = customfield_10118.get("startDate");
+				scheduledFinish = customfield_10118.get("endDate");
+				actualFinish = customfield_10118.get("completeDate");
+			}
+
+			children = new ArrayList<>();
+			JSONArray childrenJson = fields.getJSONArray("subtasks");
+			for (int i = 0; i < childrenJson.length(); i++) {
+				JSONObject subfields = childrenJson.getJSONObject(i).getJSONObject("fields");
+				JIRAIssue subIssue = resolveIssue(subfields, true, scheduledStart, scheduledFinish, actualFinish);
+				children.add(subIssue);
+			}
+			JSONObject progressObj = fields.getJSONObject("progress");
+			String progress = progressObj.getString("progress");
+			total = progressObj.getString("total");
+			percentComplete = "0".equals(total) ? "0"
+					: (Double.parseDouble(progress) / Integer.parseInt(total) * 100) + "";
+		}
+		return new JIRAIssue(name, type, status, scheduledStart, scheduledFinish, null, total, null, percentComplete,
+				actualFinish, null, null, null, children);
+	}
+
+	private Map<String, String> resolveCustomfield_10118(String customfield_10118) {
```

Comment:
Can you add some comments to explain what you're doing here? I have no idea by reading code :(